src/github.com/matrix-org/matrix-federation-tester/main.go

```diff
@@ -147,6 +147,16 @@ func Report(
 	// Map of network address to connection error.
 	report.ConnectionErrors = make(map[string]error)
 	now := time.Now()
+
+	// Retrieve domain name to send as SNI to server
+	sni, _, valid := gomatrixserverlib.ParseAndValidateServerName(serverHost)
+	if !valid {
+		parseErr := errors.New(fmt.Sprintf("Invalid serverHost %s", serverHost))
```

Comment:
it's not a `serverHost` to anyone but you:

```suggestion
		parseErr := errors.New(fmt.Sprintf("Invalid server name %q", serverHost))
```